src/github_repository.py

```diff
@@ -2,13 +2,34 @@
     Auth,
     BadCredentialsException,
     Github,
+    GitRelease,
 )
 
 
 class GithubRepository:
 
-    def __init__(self, api_token: str) -> None:
-        self.github_account = self._get_github_account(api_token)
+    def __init__(self, api_token: str, organizacion_name: str, repository_name: str) -> None:
+
+        self.api_token = api_token
+        self.organizacion_name = organizacion_name
+        self.repository_name = repository_name
+
+        self.github_account = self._get_github_account(self.api_token)
+        self.repository = self.github_account.get_repo(f'{self.organizacion_name}/{self.repository_name}')
+
+        last_release = self._get_last_release()
+        commit_sha = self._get_commit_sha_of_release(last_release)
+        commit_short_sha = commit_sha[:7]
+
+        self.version_info = {
+            "last_release": last_release,
+            "commit_sha": commit_sha,
+            "commit_short_sha": commit_short_sha,
+        }
```

Comment:
Solo un comentario menor y casi sin importancia. 
Dejaría __init__ para inicializar dependencias o variables necesarias de la instancia, como por ejemplo self.api_token, self.organization_name o self.repository.
self.version_info quizá puede ser movido a un metodo get_version_info. De esa forma evitas dejas cada metodo con un proposito mas definido.

What do you think?

Reply:
Me aprece bien lo de sepearar mejor los propositos wrappeando esta logica. Este metodo `get_version_info` vos estas pensando llamarlo desde el init tambien ? POr que creo que estaria bueno continuar creando el version info en el init. Que  pensas?

Reply:
Que ganamos generando la version info en el init? Creo que el init es mas para inicializar dependencias y variables que sean fundamentales para el funcionamiento de la class GithubRepository. Luego si hay algun comportamiento, lo dejaría para los metodos... 
Si alguien necesita la _version_info_, que la pida, es como un valor "calculado"

Reply:
Ah, bien, entiendo por que lo propones. 

> Que ganamos generando la version info en el init?

Quizas ganamos el hecho de no tener que recalcularlo ondemand mas de unas vez, osea una vez se inicializa ya esta esos valores cargados. y los metodos que lo necesitan (por ahora es uno solo) ya lo tienen.

Por ejemplo cuando se necesite guardar el `.version.yaml` se va a necesitar esta info, y me parece mas consistente que use el que ya calculo que volver a calcularlo (pensando en un futuro que esto va a ser automatico sin un operario) ya que si vuelve a calcularse yjusto se hizo un fix o un nuevo release o commit puede haber un caso bordisimo ajaja